allennlp/models/heads/classifier_head.py

```diff
@@ -0,0 +1,136 @@
+from typing import Dict, Optional
+
+from overrides import overrides
+import torch
+
+from allennlp.data import Vocabulary
+from allennlp.models.heads.head import Head
+from allennlp.models.model import Model
+from allennlp.modules import FeedForward, Seq2VecEncoder
+from allennlp.modules.seq2vec_encoders import ClsPooler
+from allennlp.training.metrics import CategoricalAccuracy
+
+
+@Head.register("classifier")
+class ClassifierHead(Model):
+    """
+    A classification `Head`.  Takes encoded text, gets a single vector out of it, runs an optional
+    feedforward layer on that vector, then classifies it into some label space.
+
+    Registered as a `Head` with name "classifier".
+
+    # Parameters
+
+    vocab : `Vocabulary`
+        Used to get the number of labels, if `num_labels` is not provided, and to translate label
+        indices to strings in `make_output_human_readable`.
+    seq2vec_encoder : `Seq2VecEncoder`, optional (default = `ClsPooler`)
+        The input to this module is assumed to be a sequence of encoded vectors.  We use a
+        `Seq2VecEncoder` to compress this into a single vector on which we can perform
+        classification.  If nothing is provided, we will use a `ClsPooler`, which simply takes the
+        first element of the sequence as the single vector (and is the standard thing to do when you
+        are running a classifier on top of a transformer).
```

Comment:
We also have a `BertPooler`, which (I think) does the same thing, but benefits from pre-trained weights.

I guess for generality, we have to have `ClsPooler` here as a default, though a lot of the time `BertPooler` would be the better option.

Reply:
There's one pretrained 768x768 linear transform followed by a tanh that you're missing out on if you use a `ClsPooler` instead of a `BertPooler`.  I'd guess the effect of the transform is very small, if there is any, but that's just a guess at this point.  But you're right that we can't do that as a default, because we don't have a model name here.